hal/targets/hal/TARGET_Freescale/TARGET_KSDK2_MCUS/TARGET_K64F/storage_driver.c

```diff
@@ -202,6 +312,14 @@ static const ARM_DRIVER_VERSION version = {
     .drv = ARM_DRIVER_VERSION_MAJOR_MINOR(1,00)
 };
 
+
+#if ((!defined(CONFIG_HARDWARE_MTD_ASYNC_OPS) || CONFIG_HARDWARE_MTD_ASYNC_OPS) && \
```

Comment:
configuration_store.c uses the following symbol for enabling async ops:

```STORAGE_DRIVER_CONFIG_HARDWARE_MTD_ASYNC_OPS```

We should adopt a common symbol that makes sense across both mbed-os configuration-store and mbedmicro/mbed. I suggest :
 - prepending STORAGE_ to make it clear this is part of the storage feature
- prepending DRIVER_ to make it clear this is enabling a feature of the STORAGE_DRIVER, which is an agreed name.